modules/nextrollIdSystem.js

```diff
@@ -0,0 +1,50 @@
+/**
+ * This module adds Nextroll ID to the User ID module
+ * The {@link module:modules/userId} module is required
+ * @module modules/nextrollIdSystem
+ * @requires module:modules/userId
+ */
+
+import { deepAccess } from '../src/utils.js';
+import { submodule } from '../src/hook.js';
+import { getStorageManager } from '../src/storageManager.js';
+
+const NEXTROLL_ID_LS_KEY = 'dca0.com';
+const KEY_PREFIX = 'AdID:'
+
+export const storage = getStorageManager();
+
+/** @type {Submodule} */
+export const nextrollIdSubmodule = {
+  /**
+   * used to link submodule with config
+   * @type {string}
+   */
+  name: 'nextroll',
+
+  /**
+   * decode the stored id value for passing to bid requests
+   * @function
+   * @return {{nextroll: string} | undefined}
+   */
+  decode(value) {
+    return value;
+  },
+
+  /**
+   * performs action to obtain id and return a value.
+   * @function
+   * @param {SubmoduleConfig} [config]
+   * @returns {{id: {nextroll: string} | undefined}}
+   */
+  getId(config) {
+    const key = KEY_PREFIX + deepAccess(config, 'params.partnerId', 'undefined');
+    const dataString = storage.getDataFromLocalStorage(NEXTROLL_ID_LS_KEY) || '{}';
+    const data = JSON.parse(dataString);
+    const nextrollId = deepAccess(data, `${key}.value`);
+
+    return { id: nextrollId ? {nextroll: nextrollId} : undefined };
```

Comment:
I don't think this is the return object you want, considering the `decode()` method is just returning whatever you set here. the object should have a top level key for your id module name: `nextrollId` then the value is typically just the ID string itself.

```suggestion
    return nextrollId ? { nextrollId: nextrollId } : undefined;
```

Reply:
This suggestion is wrong. See https://github.com/Prebid/Prebid.js/blob/3fd2bb57a4c849b0cba573b6f3ddc2b65f41e76a/modules/userId/index.js#L660

The return object is not stored as is, the `object.id` is what gets stored.